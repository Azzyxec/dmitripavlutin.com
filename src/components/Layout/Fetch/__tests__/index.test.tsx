import * as React from 'react';
import { shallow } from 'enzyme';

import { LayoutFetch } from '../index';

const siteInfo = {
  title: 'Dmitri Pavlutin blog',
  description: 'Posts by Dmitri Pavlutin about software development',
  url: 'https://dmitripavlutin.com',
  repositoryUrl: 'https://github.com/panzerdp/dmitripavlutin.com',
};

const authorInfo = {
  name: 'Dmitri Pavlutin',
  description: 'Dmitri Pavlutin is a software developer specialized in Frontend technologies',
  profiles: {
    stackoverflow: 'https://stackoverflow.com/users/1894471/dmitri-pavlutin',
    twitter: 'https://twitter.com/panzerdp',
    linkedin: 'https://www.linkedin.com/in/dmitri-pavlutin/',
    github: 'https://github.com/panzerdp',
    facebook: 'https://www.facebook.com/dmitri.pavlutin',
  },
  nicknames: {
    twitter: '@panzerdp',
  },
};

const authorProfilePicture = {
  width: 100,
  height: 100,
  base64: 'base64-encoded-string',
  src: 'http://images.com/image',
  srcSet: 'some srcset values',
};

const props = {
  data: {
    site: {
      siteMetadata: {
        siteInfo,
        authorInfo,
      },
    },
    file: {
      childImageSharp: {
        resolutions: authorProfilePicture,
      },
    },
  },
};

describe('<LayoutFetch />', function() {
  it('should render its children', function() {
    const child = <div>Child</div>;
    const wrapper = shallow(<LayoutFetch {...props}>{child}</LayoutFetch>);
    expect(wrapper.contains(child));
  });

  it('should render the layout container', function() {
    const wrapper = shallow(<LayoutFetch {...props}>Child</LayoutFetch>);
    expect(wrapper).toMatchSnapshot();
  });
});
