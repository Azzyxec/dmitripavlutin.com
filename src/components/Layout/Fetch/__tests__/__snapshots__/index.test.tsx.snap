// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutFetch /> should render the layout container 1`] = `
<LayoutContainer
  authorInfo={
    Object {
      "description": "Dmitri Pavlutin is a software developer specialized in Frontend technologies",
      "email": "mail@mail.com",
      "name": "Dmitri Pavlutin",
      "nicknames": Object {
        "twitter": "panzerdp",
      },
      "profiles": Object {
        "facebook": "https://www.facebook.com/dmitri.pavlutin",
        "github": "https://github.com/panzerdp",
        "linkedin": "https://www.linkedin.com/in/dmitri-pavlutin/",
        "stackoverflow": "https://stackoverflow.com/users/1894471/dmitri-pavlutin",
        "twitter": "https://twitter.com/panzerdp",
      },
    }
  }
  authorProfilePicture={
    Object {
      "base64": "base64-encoded-string",
      "height": 100,
      "src": "http://images.com/image",
      "srcSet": "some srcset values",
      "srcSetWebp": "src-set-webp",
      "srcWebp": "src-webp",
      "width": 100,
    }
  }
  siteInfo={
    Object {
      "description": "Posts by Dmitri Pavlutin about software development",
      "repositoryUrl": "https://github.com/panzerdp/dmitripavlutin.com",
      "title": "Dmitri Pavlutin blog",
      "url": "https://dmitripavlutin.com",
    }
  }
>
  Child
</LayoutContainer>
`;
